metaflow/plugins/kfp/kfp.py

```diff
@@ -1,316 +1,395 @@
-
-
-def create_command_templates_from_graph(graph):
-    """
-    Create a map of steps to their corresponding command templates. These command templates help define the command
-    to be used to run that particular step with placeholders for the `run_id` and `datastore_root` (location of the datastore).
-
-    # Note:
-    # Level-order traversal is adopted to keep the task-ids in line with what happens during a local metaflow execution.
-    # It is not entirely necessary to keep this order of task-ids if we are able to point to the correct input-paths for
-    # each step. But, using this ordering does keep the organization of data in the datastore more understandable and
-    # natural (i.e., `start` gets a task id of 1, next step gets a task id of 2 and so on with 'end' step having the
-    # highest task id. So the paths in the datastore look like: {run-id}/start/1, {run-id}/next-step/2, and so on)
-    """
-
-    def build_cmd_template(step_name, task_id, input_paths):
+        pipeline_func = self.create_kfp_pipeline_from_flow_graph()
+        run_pipeline_result = self._client.create_run_from_pipeline_func(
+            pipeline_func,
+            arguments={},
+            experiment_name=experiment_name,
+            run_name=run_name,
+            namespace=self.namespace,
+        )
+        return run_pipeline_result
+
+    def create_kfp_pipeline_yaml(self, pipeline_file_path=DEFAULT_KFP_YAML_OUTPUT_PATH):
         """
-        Returns the python command template to be used for each step.
-
-        This method returns a string with placeholders for `datastore_root` and `run_id`
-        which get populated using the provided config and the kfp run ID respectively.
-        The rest of the command string is populated using the passed arguments which are known before the run starts.
-
-        An example constructed command template (to run a step named `hello`):
-        "python downloaded_flow.py --datastore s3 --datastore-root {ds_root} " \
-                         "step hello --run-id {run_id} --task-id 2 " \
-                         "--input-paths {run_id}/start/1"
+        Creates a new KFP pipeline YAML using `kfp.compiler.Compiler()`. Note: Intermediate pipeline YAML is saved
+        at `pipeline_file_path`
+        """
+        import kfp
+
+        pipeline_func = self.create_kfp_pipeline_from_flow_graph()
+        kfp.compiler.Compiler().compile(pipeline_func, pipeline_file_path)
+        return os.path.abspath(pipeline_file_path)
+
+    @staticmethod
+    def _command(code_package_url, environment, step_name, step_cli):
+        commands = environment.get_package_commands(code_package_url)
+
+        commands.extend(
+            [
+                "git clone -b alexla/AIP-1676 https://github.com/alexlatchford/pipelines",
+                "export oldpath=`pwd`",
+                "cd pipelines/sdk/python",
+                "pip3 install -e .",  # TODO: pip install kfp when fixed
+                "cd $oldpath",
```

Comment:
Looks like you should be able to use the `subdirectory` feature of `pip install`, see https://pip.pypa.io/en/stable/reference/pip_install/#vcs-support

Should make this into a one-liner :)